### ISSUE TITLE
Identify scope with respect to environment setup

### ISSUE THREAD

**Issue by ganeshrn** [ansible/ansible-lint#1830]
##### Summary

Reconsider scope of ansible-lint with respect to environment  setup as a pre-requisite. Currently all the dependency are installed by ansible-lint in an isolated user environment. The behaviour can be defaulted to assume the user has everything installed already (offline mode). 

Note: ansible-navigator will have support to install collections in future.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible-lint --version

```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behaviour

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behaviour

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Changing default to be `offline: true` would likely break 90% of users of the linter, as they rely on automatically installing their `requirements.yml`. On the other hand I do understand that if ansible-lint is used inside a container (execution environment), probably we do want to use `offline: true`, which would also mean that when dependencies are updated, it would be up to the user to rebuild the ee.

In our `creator-ee` we could add a root `.ansible-lint` which enables the offline true but that has two problems:

* Some people will not be able to use creator-ee directly, w/o having to build their custom container on top of it.
* If their code already contains an `ansible-lint config, that would take precedence over the root config, basically undoing the offline mode enabling.

So, for the moment, those wanting to use offline mode, should just put `offline: true` in the config.

**Comment by ganeshrn** [comment]
Closing the issue as we will keep the default behaviour as it is.
